paasta_tools/cli/cmds/status.py

```diff
@@ -385,8 +385,19 @@ def apply_args_filters(args):
 
     filters = get_filters(args)
 
+    all_services = list_services(soa_dir=args.soa_dir)
+
+    if args.service and args.service not in all_services:
+        paasta_print(PaastaColors.red(f'The service "{args.service}" does not exist.'))
+        suggestions = difflib.get_close_matches(args.service, all_services, n=5, cutoff=0.5)
+        if suggestions:
+            paasta_print(PaastaColors.red(f'Did you mean any of these?'))
+            for suggestion in suggestions:
+                paasta_print(PaastaColors.red(f'  {suggestion}'))
+        return {}
+
     i_count = 0
-    for service in list_services(soa_dir=args.soa_dir):
+    for service in all_services:
```

Comment:
It seems weird to me that this thing is so flexible enough to allow multiple services like this.

Do we still need to loop like this?

Reply:
It looks like we actually always only have one service (`args.service` is always set by the code above, this is probably a remnant from some time when it wasn't). I'll try to clean this up.

Reply:
actually we don't always have a service, e.g. if you do `paasta status -c norcal-prod -i main --owner webcore`, so it probably really is necessary to loop like this.

Reply:
Ah